bash/utils.sh

```diff
@@ -335,3 +335,48 @@ function archive {
     pg_dump ${BUILD_ENGINE} -t ${src} -O -c | sed "s/${src}/${dst}/g" | psql ${EDM_DATA}
     psql ${EDM_DATA} -c "COMMENT ON TABLE ${dst} IS '${DATE} ${commit}'"
 }
+
+
+function docker_login() {
+    echo "$DOCKER_PASSWORD" | docker login -u $DOCKER_USER --password-stdin
+}
+
+
+function docker_tag_exists() {
+    docker manifest inspect $1:$2 >/dev/null
+}
+
+
+function build_and_publish_docker_image {
```

Comment:
rationale for these two is that the bottom one is likely better practice - if we really have a more rigid versioning (like geosupport), would be good to not change an image once it's published. But that being said, sometimes flexibility is good so I have this one too.